src/main/java/it/gov/pagopa/payhub/activities/activity/paymentsreporting/PaymentsReportingIngestionFlowFileActivityImpl.java

```diff
@@ -57,15 +57,15 @@ protected PaymentsReportingIngestionFlowFileActivityResult handleRetrievedFiles(
 		List<PaymentsReporting> paymentsReportings = parseData(retrievedFiles.getFirst().toFile(), ingestionFlowFileDTO);
 		paymentsReportingService.saveAll(paymentsReportings);
 
-		List<TransferSemanticKeyDTO> transferSemanticKeys = paymentsReportings.stream()
-			.map(paymentsReportingMapperService::map2TransferSemanticKeyDto)
-			.toList();
-		return new PaymentsReportingIngestionFlowFileActivityResult(transferSemanticKeys, true, null);
+		List<TransferSemanticKeyWithOutComeCodeDTO> transferSemanticKeys = paymentsReportings.stream()
+			.map(paymentsReportingMapperService::map2TransferSemanticKeyWithOutComeCodeDTO).toList();
+
+		return new PaymentsReportingIngestionFlowFileActivityResult(paymentsReportings.getFirst().getIuf(), transferSemanticKeys, true, null);
```

Comment:
let's document with a comment that the iuf is always the same ;)
move the iuf declaration after retrieving the list of paymentsReporting
```suggestion
		String iuf = paymentsReportings.getFirst().getIuf(); // The iuf is the same for entire file
		return new PaymentsReportingIngestionFlowFileActivityResult(iuf, transferSemanticKeys, true, null);
```